### PR TITLE
[cmd] status: log at info level always when running the command

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

This PR sets the log level when running the agent status command always to
`info` as that is the level at which the status is logged to screen.

### Motivation

Currently the log level for the status command is grabbed from the environment
or defaults to `info`. The feature works as expected on hosted environments, but
can break the feature on containerized environments due to the DD_LOG_LEVEL
env var being commonly used. We should _always_ default to log level for this
command as the entire output from the status map is logged using `log.Info(...)`. 

### Additional Notes

None

### Possible Drawbacks / Trade-offs

None

### Describe how to test/QA your changes

1. Start a containerized agent and set `DD_LOG_LEVEL` to "warn" or higher.
2.  Run the agent status command: `agent status`
3. Make sure there's correct valid status update; as opposed to nothing.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ x At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by olivielpeau** [comment]
Why is the output of the `status` command logged instead of `fmt.Print`ed out to stdout? I'd expect a CLI command to simply print out to the console directly the output it wants to print out instead of going through the logger 🤔

**Review by sgnn7 — APPROVED** [review]

**Comment by sgnn7** [comment]
@olivielpeau It's to make sure the output is scrubbed. The PR for the original change in question is [here](https://github.com/DataDog/datadog-agent/pull/12338).

**Review by scottopell — APPROVED** [review]

File: releasenotes/notes/agent-status-log-level-fix-93b84dd50b4a9257.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Agent status command should always log at info level to avoid
+    interference from agent-wide log-level setting in printout.
```
Comment (buraizu) [inline releasenotes/notes/agent-status-log-level-fix-93b84dd50b4a9257.yaml:12]:
```suggestion
    Agent status command should always log at info level to allow
    full status output regardless of Agent log level settings.
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks, just a minor suggestion for clarity

**Review by buraizu — APPROVED** [review]

**Pull request merged**